File: src/lemon.c
#include <stdio.h>

#include "include/parser.h"
#include "include/token.h"
#include "include/io.h"
#include "include/compiler.h"

int main(int argc, char *argv[])
{
    if (argv[1] == NULL)
    {
        printf("[lemonc] Please provide a file to compile.");
        return 1;
    }
    printf("[lemonc] Start compiling...\n");
    lexer_T *lexer = init_lexer(io_read_file(argv[1]));
    parser_T *parser = init_parser(lexer);
    AST_T *structure = parser_parse(parser);
    free(lexer);
    free(parser);
    io_write_file("lemon_application.asm", compile_windows_32(structure));
    free(structure);
    printf("[lemonc] Finished compiling! :D\n");
}

File: src/Lemon.cs
using LemoncNS;

class Lemonc
{
    static int Main(string[] args)
    {
        if (args.Length != 2)
        {
            sendHelp();
        }
        else if (args[0] == "simulate")
        {
            Console.WriteLine("[lemonc] Start simulating...");
            Lexer lexer = new Lexer(System.IO.File.ReadAllText(args[1]));
            Parser parser = new Parser(lexer);
            AST structure = parser.parse();
            Interpreter.interpret(structure);
            return 0;
        }
        else if (args[0] == "compile")
        {
            Console.WriteLine("[lemonc] Start compiling...");
            Lexer lexer = new Lexer(System.IO.File.ReadAllText(args[1]));
            Parser parser = new Parser(lexer);
            AST structure = parser.parse();
            System.IO.File.WriteAllText("lemon_application.asm", Compiler.compileFasmWindows32(structure));
            return 0;
        }
        else
        {
            sendHelp();
        }
        return 1;
    }

    static void sendHelp()
    {

        Console.WriteLine("[lemonc] Wrong arguments.");
        Console.WriteLine("[lemonc] Expected:");
        Console.WriteLine("[lemonc] => lemonc simulate <filename>");
        Console.WriteLine("[lemonc] => lemonc compile <filename>");
    }
}
